app/src/main/java/com/github/openwebnet/repository/impl/EnvironmentRepositoryImpl.java

```diff
@@ -102,4 +106,19 @@ public EnvironmentRepositoryImpl() {
         });
     }
 
+    @Override
+    public Observable<Void> delete(Integer id) {
+        return Observable.create(subscriber -> {
+            try {
```

Comment:
As expected, both

```java
databaseRealm.delete(DomoticModel.class, DomoticModel.FIELD_ENVIRONMENT_ID, id);
```
and 
```java
databaseRealm.delete(RealmObject.class, DomoticModel.FIELD_ENVIRONMENT_ID, id);
```
don't work. The first does not compile, the second throws at runtime.

I you'd still like to use reflection, I think that [this project](https://github.com/ronmamo/reflections) could help, but there's no Android integration.

Anyway, I still think using reflection would be overkill here.